cfme/infrastructure/virtual_machines.py

```diff
@@ -1071,113 +1174,118 @@ class VmAllWithTemplatesDetails(CFMENavigateStep):
     VIEW = InfraVmDetailsView
     prerequisite = NavigateToSibling('All')
 
-    def step(self, *args, **kwargs):
-        sel.click(self.obj.find_quadicon(do_not_navigate=True))
+    def step(self):
+        self.prerequisite_view.sidebar.vmstemplates.tree.click_path('All VMs & Templates')
```

Comment:
this should be in TODO - override VmEntities to provide - provider and type / search not only by name. 
Here it is used that way to have 1 navigation step for both VM and Templates. with has some gaps - we do not check if name is unique